packages/driver/src/cy/assertions.coffee

```diff
@@ -328,6 +351,10 @@ create = (state, queue, retryFn) ->
       message = message.replace(stackTracesRe, "\n")
 
     obj = parseValueActualAndExpected(value, actual, expected)
+    
+    if showDiff(error)
+      diffObjs = prepareObjsForDiff(obj)
+      diff = JsDiff.createPatch('string', diffObjs.actual, diffObjs.expected)
```

Comment:
the condition inside `prepareObjsForDiff` is OR `if _.isString(err.actual) || _.isString(err.expected)`. So will `JsDiff` do a good job when actual is a string, and expected is an object?

Reply:
no diff will be shown when comparing string to object, since showDiff(error) would return false